spec/amp-shadow-doc.md

```diff
@@ -0,0 +1,104 @@
+<!---
+Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# AMP Shadow Doc API
+
+## Overview
+
+AMP can be used in a shadow-doc mode where a single web page can open many AMP
+documents. This is especially useful for shell-style PWA documents.
+
+## Using shadow doc API
+
+The special runtime should be used in place of `v0.js`. It can be declared in the
+shell page as following:
+
+```html
+<script async src="https://cdn.ampproject.org/shadow-v0.js"></script>
+
+<!-- Wait for API to initialize -->
+<script>
+(window.AMP = window.AMP || []).push(function(AMP) {
+  // AMP APIs can be used now via "AMP" object.
+});
+</script>
+```
+
+## API
+
+### Fetching and attaching shadow docs
+
+There are currently two ways how one can attach a shadow doc: using a `Document` object, loaded, for instance, via XHR. Or using an experimental streaming API. Once streaming API graduates from experimental, the non-streaming API will be deprecated.
+
+Using the fully loaded `Document` object:
+```javascript
+fetchDocumentViaXhr(url).then(fetchedDoc => {
+  const shadowDoc = AMP.attachShadowDoc(hostElement, fetchedDoc, url, options);
+});
+```
+
+Using the streaming API:
+```javascript
+const shadowDoc = AMP.attachShadowDocAsStream(hostElement, url, options);
+fetch(url).then(response => {
```

Comment:
Possibly. I'll consider for a follow up PR. This API, however, looks the best to me so far on the generic level.